core/DbHelper.php

```diff
@@ -352,6 +352,32 @@ public static function addOriginHintToQuery(string $sql, string $origin, ?Date $
         return $sql;
     }
 
+    /**
+     * Add an optimizer hint to the query to set the first table used by the MySQL join execution plan
+     *
+     * https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html#optimizer-hints-join-order
+     *
+     * @param string $sql       SQL query string
+     * @param string $prefix    Table prefix to be used as the first table in the plan
+     *
+     * @return string           Modified query string with hint added
+     */
+    public static function addJoinPrefixHintToQuery(string $sql, string $prefix): string
+    {
+        $generalConfig = Config::getInstance()->General;
+        if (empty($generalConfig['enable_segment_first_table_join_prefix']) || $generalConfig['enable_segment_first_table_join_prefix'] != "1") {
+            return $sql;
+        }
+
+        $select = 'SELECT';
+        if (0 === strpos(trim($sql), $select)) {
+            $sql = trim($sql);
+            $sql = 'SELECT /*+ JOIN_PREFIX(' . $prefix . ') */' . substr($sql, strlen($select));
+        }
+
+        return $sql;
```

Comment:
Not the most pressing thing here, just a code shortening thing.

We can either do an early return like we have here or when a thing is dependent on a config being set and enabled, this condition check is shorter and covers both the 'is set' and 'is truthy' as the null coalesce handles the array key not set as well, and we can drop the variable.

```suggestion
        if (Config::getInstance()->General['enable_segment_first_table_join_prefix'] ?? false) {
            $select = 'SELECT';
            if (0 === strpos(trim($sql), $select)) {
                $sql = trim($sql);
                $sql = 'SELECT /*+ JOIN_PREFIX(' . $prefix . ') */' . substr($sql, strlen($select));
            }
        }

        return $sql;
```

Reply:
The config check has now been moved elsewhere, but thanks for the cleaner code suggestion :+1: